### ISSUE TITLE
Does not resolve dependencies with security groups

### ISSUE THREAD

**Issue by Spechal** [terraform-aws-modules/terraform-aws-security-group#47]
```
module "public_subnet_sg" {
  source = "terraform-aws-modules/security-group/aws"

  name        = "${var.environment}-public-subnets"
  description = "Security group for public facing services"
  vpc_id      = "${module.vpc.vpc_id}"

  ingress_with_self = [
    {
      rule = "all-all"
    },
  ]

  ingress_with_cidr_blocks = [
    {
      from_port   = 80
      to_port     = 80
      protocol    = "tcp"
      description = "default http port"
      cidr_blocks = "0.0.0.0/0"
    },
  ]

  ingress_with_source_security_group_id = [
    {
      from_port                = 22
      to_port                  = 22
      protocol                 = "tcp"
      description              = "SSH"
      source_security_group_id = "${module.bastion_sg.this_security_group_id}"
    },
  ]

  egress_cidr_blocks = ["0.0.0.0/0"]
}

module "bastion_sg" {
  source = "terraform-aws-modules/security-group/aws"

  name        = "${var.environment}-bastion"
  description = "Security group for bastion"
  vpc_id      = "${module.vpc.vpc_id}"

  ingress_with_cidr_blocks = [
    {
      rule = "all-all"
    },
  ]

  egress_cidr_blocks = ["0.0.0.0/0"]
}
```

```
* module.public_subnet_sg.aws_security_group_rule.ingress_with_source_security_group_id: aws_security_group_rule.ingress_with_source_security_group_id: value of 'count' cannot be computed
```

```
Terraform v0.11.2
+ provider.archive v0.1.0
+ provider.aws v1.8.0
+ provider.null v1.0.0
```

If you remove the ingress rules to create the group, add them back and re-apply things work as expected.

**Comment by antonbabenko** [comment]
There is a note about this issue in [README.md](https://github.com/terraform-aws-modules/terraform-aws-security-group#1-security-group-with-custom-rules).

**Comment by antonbabenko** [comment]
This is now fixed in [v2.0.0](https://github.com/terraform-aws-modules/terraform-aws-security-group/releases/v2.0.0). See this section in [README](https://github.com/terraform-aws-modules/terraform-aws-security-group#note-about-value-of-count-cannot-be-computed)
